kernel/kernel-api/src/main/java/io/delta/kernel/Snapshot.java

```diff
@@ -148,24 +145,25 @@ enum ChecksumWriteMode {
   /**
    * Writes a checksum file for this snapshot using the specified mode.
    *
+   * <p>This method should only be called if a checksum file does not already exist at this version.
```

Comment:
what is the expected behavior if it does already exist?

i.e. two snapshots created, neither thinks it exists, both try to write it

Maybe (a) call out in the docs and (b) test for this?